implementations/rust/ockam/ockam/src/abac/access_control.rs

```diff
@@ -0,0 +1,60 @@
+//! [`AccessControl`] implementation which uses [`AbacAuthorization`]
+//! to resolve authorization requests.
+
+use core::fmt::{self, Debug, Formatter};
+
+use crate::abac::{AbacAuthorization, AbacLocalInfo};
+use ockam_core::async_trait;
+use ockam_core::compat::boxed::Box;
+use ockam_core::{AccessControl, LocalMessage, Result};
+
+/// Allows only messages that pass attribute based access control.
+pub struct AttributeBasedAccessControl<A> {
+    backend: A,
+}
+
+impl<A> Debug for AttributeBasedAccessControl<A> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "AttributeBasedAccessControl")
+    }
+}
+
+impl<A> AttributeBasedAccessControl<A> {
+    /// Create a new `AttributeBasedAccessControl` with the given
+    /// [`AbacAuthorization`] backend.
+    pub fn new(backend: A) -> Self
+    where
+        A: AbacAuthorization + Debug + Send + Sync + 'static,
+    {
+        Self { backend }
+    }
+
+    /// Return a reference to the [`AbacAuthorization`] implementation.
+    pub fn backend(&self) -> &A {
+        &self.backend
+    }
+}
+
+#[async_trait]
+impl<A> AccessControl for AttributeBasedAccessControl<A>
+where
+    A: AbacAuthorization + Debug + Send + Sync + 'static,
+{
+    async fn is_authorized(&self, local_msg: &LocalMessage) -> Result<bool> {
+        // pull request triple out of LocalMessage's LocalInfo
+        let local_info = AbacLocalInfo::find_info(local_msg)?;
```

Comment:
We don't have to use it, but I can think of use-cases where it could be useful for customers. If you'd prefer I can move it to the ockam_examples/ section.

Reply:
Moved to `ockam_examples/example_projects/access_control`.